### PR TITLE
Add support of NVARCHAR data type

### PR DESCRIPTION
**Pull request by yuval-illumex**
Redshift support the NVARCHAR datatype ([Redshift Data Types](https://docs.aws.amazon.com/redshift/latest/dg/c_Supported_data_types.html))

The following query can't be parsed using sqlparser:

```
SELECT cast(field as nvarchar(50)) as "field" 
FROM a

```

### CODE REVIEW

File: src/ast/data_type.rs [hunk 1]
```diff
@@ -98,6 +100,9 @@ impl fmt::Display for DataType {
             DataType::Varchar(size) => {
                 format_type_with_optional_length(f, "CHARACTER VARYING", size, false)
             }
+            DataType::Nvarchar(size) => {
+                format_type_with_optional_length(f, "Numeric CHARACTER VARYING", size, false)
```
Comment (alamb) [inline src/ast/data_type.rs:104]:
```suggestion
                format_type_with_optional_length(f, "NUMERIC CHARACTER VARYING", size, false)
```

I think this should be all capitals to be consistent with the rest of this function

**Review by alamb — COMMENTED** [review]
Thanks @yuval-illumex  -- I have one small suggestion and I think this PR needs a test. Otherwise it should be good to go

**Comment by yuval-illumex** [comment]
Thanks for the feedback @alamb

**Comment by alamb** [comment]
Thanks @yuval-illumex  -- can you please add a test (so that someone doesn't accidentally break this feature in the future and not know it?)

**Review by alamb — APPROVED** [review]
Thanks -- looks good @yuval-illumex

**Pull request merged**